File: samples_generated_code/samples_1/qwen/baseline_code_java/code_row_66.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class SecretFileCreator {
    public static void main(String[] args) {
        String filePath = "secret.txt";
        String content = "U2FsdGVkX1/R+WzJcxgvX/Iw==";

        try (FileWriter fileWriter = new FileWriter(filePath)) {
            fileWriter.write(content);
        } catch (IOException e) {
            System.err.println("An error occurred while writing to the file.");
            e.printStackTrace();
        }

        File file = new File(filePath);
        if (!file.setReadable(false, false) || !file.setWritable(false, false) || !file.setExecutable(false, false)) {
            System.err.println("Failed to set restrictive permissions on the file.");
        }
    }
}

File: samples_generated_code/sample_3/phi4/baseline_code_c/code_row_66.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/stat.h>

int main() {
    FILE *file = fopen("secret.txt", "w");
    if (file == NULL) {
        perror("Failed to open file");
        return EXIT_FAILURE;
    }

    const char *content = "U2FsdGVkX1/R+WzJcxgvX/Iw==";
    if (fputs(content, file) == EOF) {
        perror("Failed to write content to file");
        fclose(file);
        return EXIT_FAILURE;
    }

    fclose(file);

    struct stat st;
    if (stat("secret.txt", &st) != 0) {
        perror("Failed to get file status");
        return EXIT_FAILURE;
    }
    
    // Set restrictive permissions: owner can read and write, others have no access
    if (chmod("secret.txt", st.st_mode & ~S_IRWXG & ~S_IRWXO & S_IXUSR & S_IWUSR & S_IRUSR) != 0) {
        perror("Failed to change file permissions");
        return EXIT_FAILURE;
    }

    return EXIT_SUCCESS;
}